### ISSUE TITLE
Refactor code to make things more modular

### ISSUE THREAD

**Issue by jonathontoon** [jonathontoon/manifest#56]
## Expected Behavior
Make things easier to read and follow

## Current Behavior
A single file performs well but is hard to parse.

**Comment by ibrahimbutt** [comment]
@jonathontoon have you considered using TypeScript? It could help in making things more modular, as it allows for better OOP, if you choose to go the OOP route that is.

Besides that, have you considered testing? I would be happy to help in that regard, and thereon with refactoring.

**Comment by jonathontoon** [comment]
@ibrahimbutt I think Typescript, while I personally like aspects of it would close off a lot of people with only fundamental Javascript knowledge who might have been able to contribute otherwise.

Testing should come a long at some point. As I'm mapping out what Manifest 2.0 might look like I think it should be considered at the start. I've personally only ever used basic snapshot testing for React components and Jest for Redux. Where might be a good place to start test-wise with vanilla Javascript?

**Comment by ankitatechie** [comment]
@jonathontoon I think there is a great scope in CSS improvements as well. Right now selectors are tightly coupled to a specific HTML structure along with deep level nesting.

If you prefer, I can fix it up for you. Also, I can help you with making it more presentable with minimal code.
